### PR TITLE
Adding libxml2 xmlAddChild binding to XML standard library

### PR DESCRIPTION
**Pull request by Vici37**
### What
Adding the libxml2 xmlAddChild binding to the XML node class so that XML nodes can ... add children.

### Why
I'm working on a personal project involving some XML that requires some nodes being shuffled around, and couldn't find an approach to move child nodes from one parent node to another. Please let me know if I'm just blind and there's a completely valid approach to doing this.

### CODE REVIEW

File: src/xml/node.cr [hunk 1]
```diff
@@ -477,6 +477,11 @@ struct XML::Node
     LibXML.xmlUnlinkNode(self)
   end
 
+  # Adds the node as the child of this node in the XML document
+  def addChild(child : Node)
```
Comment (r00ster91) [inline src/xml/node.cr]: In Crystal (or in the standard library at least) methods have the `snake_case`, not the `camelCase`: (spec needs to be changed too)
```suggestion
  def add_child(child : Node)
```

Comment (Vici37) [reply src/xml/node.cr]: Good catch! Fixed it up. I've been spending too much time in Java land.

**Comment by asterite** [comment]
Also note that in Nokogiri (libxml2 bindings for Ruby) [this method](https://github.com/sparklemotion/nokogiri/blob/c021f65d94f979bb8656d989949fc1bfc57b91a6/ext/nokogiri/xml_node.c#L163) is not just a single line: it checks that the node is OK to be reparented, and removes the previous parent from the given node, etc. It's **very complex**. So let's be careful with this.

**Comment by Vici37** [comment]
That's fair @asterite - for my project, I'm not too concerned if I shoot myself in the foot by doing something improper. To me, it's more valuable to have the xmlAddChild method that may break me in certain edge cases than to not have xmlAddChild at all. Obviously a less than stellar experience for the standard library if something does go sideways, so I understand the desire to get it right first.

I'll work on porting over the Nokogiri implementation (thanks for the link!) and will update this PR when I've gotten through that rabbit hole.

Thanks for the feedback!

**Pull request closed without merging**